src/xml/rss.rs

```diff
@@ -35,15 +35,17 @@ pub struct Channel {
   pub last_build_date: Option<String>,
   pub pub_date: Option<String>,
   #[serde(rename = "item")]
-  pub items: Vec<RssPost>,
+  pub items: Option<Vec<RssPost>>,
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 #[serde(rename_all = "camelCase")]
 #[serde(rename = "item")]
 pub struct RssPost {
-  pub title: String,
-  pub link: String,
+  // Link and title can be omitted, according to spec, provided that there is a description
+  // https://www.rssboard.org/rss-specification#hrelementsOfLtitemgt
+  pub title: Option<String>,
+  pub link: Option<String>,
```

Comment:
While this is the case and they can be omitted, I need both of them for the app to work, which is why I did not make them optional. Although I guess you do handle these missing later on which also makes sense, nevermind.